MekHQ/src/mekhq/gui/dialog/GMToolsDialog.java

```diff
@@ -42,887 +39,1256 @@
-    /**
-     * The last unit rolled, used when clicking 'Add Random Unit' after Roll for RAT is clicked
-     */
-    private MechSummary lastRolledUnit;
+    private final CampaignGUI gui;
+    private final Person person;
 
-    /**
-     * The last name rolled, used when clicking 'Set Generated Name' after Generate Name is clicked
-     */
+    //region GUI Variables
+    private JTabbedPane tabbedPane;
+
+    //region General Tab
+    private JSpinner spnDiceCount;
+    private JSpinner spnDiceNumber;
+    private JSpinner spnDiceSides;
+    private JLabel lblTotalDiceResult;
+    private JTextPane txtIndividualDiceResults;
+
+    private MMComboBox<FactionDisplay> comboRATFaction;
+    private JTextField txtYear;
+    private MMComboBox<String> comboQuality;
+    private MMComboBox<String> comboUnitType;
+    private MMComboBox<String> comboUnitWeight;
+    private JLabel lblUnitPicked;
+    private Entity lastRolledUnit;
+    //endregion General Tab
+
+    //region Name Tab
+    private MMComboBox<String> comboEthnicCode;
+    private MMComboBox<Gender> comboGender;
+    private MMComboBox<FactionDisplay> comboNameGeneratorFaction;
+    private JCheckBox chkClanner;
+    private JSpinner spnNameNumber;
+    private JLabel lblCurrentName;
+    private JTextArea txtNamesGenerated;
     private String[] lastGeneratedName;
 
-    /**
-     * The last callsign rolled, used when clicking 'Set Generated Callsign' after Generate Callsign is clicked
-     */
+    private JSpinner spnCallsignNumber;
+    private JLabel lblCurrentCallsign;
+    private JTextArea txtCallsignsGenerated;
     private String lastGeneratedCallsign;
 
-    /**
-     * The last bloodname rolled, used when clicking 'Set Generated Bloodname' after Generate Bloodname is clicked
-     */
-    private String lastGeneratedBloodname;
-    //endregion Previously Generated Information
-
-    //region Data Sources
-    private CampaignGUI gui;
-    private Person person;
-    private List<Clan> clans = new ArrayList<>();
-    //endregion Data Sources
-
-    //region Bloodname Information
-    private int originClan;
-    private Phenotype selectedPhenotype;
+    private MMComboBox<ClanDisplay> comboOriginClan;
+    private MMComboBox<Integer> comboBloodnameEra;
+    private MMComboBox<Phenotype> comboPhenotype;
+    private JLabel lblCurrentBloodname;
+    private JLabel lblBloodnameGenerated;
+    private JLabel lblOriginClanGenerated;
+    private JLabel lblPhenotypeGenerated;
+    private JLabel lblBloodnameWarning;
+    private Clan originClan;
     private int bloodnameYear;
-    //endregion Bloodname Information
+    private Phenotype selectedPhenotype;
+    private String lastGeneratedBloodname;
+    //endregion Name Tab
+    //endregion GUI Variables
 
     //region Constants
     private static final String[] QUALITY_NAMES = {"F", "D", "C", "B", "A", "A*"};
     private static final String[] WEIGHT_NAMES = {"Light", "Medium", "Heavy", "Assault"};
```

Comment:
Just in case we didn't have enough weight and quality names scattered around MegaMek/MekHQ, lol. No need to address since it's exsiting code.